horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -1004,4 +1004,137 @@ public DatasetList listDatasetsBySchema(String uri, Integer limit, Integer page,
       DatasetList list = new DatasetList();
       return list;
    }
+
+   private void transform(Run run, String role, String token) {
+      if (run == null || role == null) {
+         log.errorf("Transformation parameters error: run %s, role %s", run, role);;
+      }
+      Collection<RunService.RunSchema> runSchemas = getSchemas(run, role, token);
+      DataSet ds = null;
+
+      try (@SuppressWarnings("unused") CloseMe h1 = roleManager.withRoles(em, role);
+            @SuppressWarnings("unused") CloseMe h2 = roleManager.withToken(em, token)) {
+
+         Test test = Test.findById(run.testid);
+         List<JsonNode> nodesToProcess = new ArrayList<JsonNode>();
+         boolean matched = false;
+   
+         if (!runSchemas.isEmpty()){
+            for ( RunSchema s : runSchemas ){
+               findMatches(s, run.data, nodesToProcess, true);
+               int nodeSize = nodesToProcess.size();
+   
+               if (nodeSize == 0) {
+                  new DataSet(run.start, run.stop, "",
+                        test.id, run.data, run, 0, run.owner, run.access)
+                        .persist();
+               } else {
+                  Collection<Transformer> transformers = test.transformers;
+                  if (transformers.isEmpty()) {
+                     new DataSet(run.start, run.stop, "",
+                           test.id, run.data, run, 0, run.owner, run.access)
+                           .persist();
+                  } else {
+                     for (Transformer t : transformers){
+                        ArrayNode root = JsonNodeFactory.instance.arrayNode();
+
+                        for (int nodePos = 0; nodePos <= nodeSize; nodePos += 1 ) {
+                           JsonNode n = nodesToProcess.get(nodePos);
+                           if (!t.extractors.isEmpty()) {
+                              for ( NamedJsonPath e : t.extractors ){
+                                 JsonNode match = n.path(e.name);
+                                 if (!match.isMissingNode()) {
+                                    ObjectNode extracted = instance.objectNode();
+                                    extracted.put(e.name, match.textValue());
+                                    root.add(extracted);
+                                    matched = true;
+                                 }
+                              }
+                           }
+                        }
+
+                        if (t.function == null || t.function.isBlank()) {
+                           ds = new DataSet(run.start, run.stop
+                              , "", test.id
+                              , root, run
+                              , 0, run.owner, run.access);
+                        } else {
+                           StringBuilder code = new StringBuilder();
+                           code.append("const __obj = { __obj == "+root.asText()+" }").append(";\n");
```

Comment:
uh, I don't understand the `==` - does this compute?

Reply:
no it won't.